packages/test/test-service-load/src/loadTestDataStore.ts

```diff
@@ -513,7 +513,7 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
         const signalsGapMs = cycleMs / signalsPerCycle;
         let submittedSignals = 0;
         try {
-            while (submittedSignals < clientSignalsSendCount) {
+            while (submittedSignals < clientSignalsSendCount && this.runtime.connected && !this.runtime.disposed) {
                 // all the clients are sending signals;
                 // with signals, there is no particular need to have staggered writers and readers
                 this.runtime.submitSignal("generic-signal", true);
```

Comment:
what will not looping do? would it be better to just wrap the submit and submitted++ call, so we will basically just delay while disconnected?

Reply:
That's valid, thanks! Making the change :")

Reply:
i think the disposed check should probably stay in the loop

Reply:
Done, thanks!